bitcoin/src/blockdata/transaction.rs

```diff
@@ -580,13 +580,27 @@ pub struct Transaction {
     ///
     /// * [BIP-65 OP_CHECKLOCKTIMEVERIFY](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)
     /// * [BIP-113 Median time-past as endpoint for lock-time calculations](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki)
-    pub lock_time: absolute::PackedLockTime,
+    pub lock_time: absolute::LockTime,
     /// List of transaction inputs.
     pub input: Vec<TxIn>,
     /// List of transaction outputs.
     pub output: Vec<TxOut>,
 }
 
+impl cmp::PartialOrd for Transaction {
+    fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl cmp::Ord for Transaction {
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
+        self.version.cmp(&other.version)
+            .then(self.lock_time.to_consensus_u32().cmp(&other.lock_time.to_consensus_u32()))
+            .then(self.input.cmp(&other.input))
+            .then(self.output.cmp(&other.output))
+    }
+}
```

Comment:
One thing I just realized: the fields are not ordered in the same order they are on the wire. Could it be a problem?

Reply:
It's been this way since the beginning of this library and nobody has complained, but I'm happy to change this to match the wire ordering just in case.

Reply:
Hard question since we could break someone's code by changing it. There isn't any standardized sort order of transactions, is there?

Reply:
No, there isn't. If I had to guess, the only specific sort order that anybody is using is one based on TXIDs, for indexing purposes.

Our `Ord` impl is purely here to make supporting ordered collections easy.